### PR TITLE
Actions & Sources - Transistor.fm

### LINKED ISSUE #3818: Transistor.fm
### App
Transistor.fm

### Triggers
* New Episode Created
* New Subscriber Created
* Episode Published
* Subscriber Removed

### Actions
* Create Subscriber
* Remove Subscriber

### PR DESCRIPTION
**Pull request by vellames**
Closes #3818 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4604"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Review by alysonturing — CHANGES_REQUESTED** [review]
Thanks for your contribution!
I've added a couple of comments below

File: components/transistor_fm/sources/new-event/new-event.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    eventName: {
+      type: "string",
+      label: "Event Name",
+      description: "The name of the event you want to subscribe to.",
+      options: options.EVENT_NAME_OPTIONS,
+    },
+  },
+  hooks: {
+    async activate() {
```
[the remark this answers is not in the document]
Comment (vellames) [reply components/transistor_fm/sources/new-event/new-event.mjs:28]: Activate is executed every time the component is deployed or updated :)

File: components/transistor_fm/sources/new-event/new-event.mjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        summary: `${this.eventName} - ${data.attributes.email}`,
+        ts: data.attributes.created_at || data.attributes.last_notified_at || new Date(),
+      });
+    },
+  },
+  async run(event) {
+    console.log("Event received");
+    console.log("WARN: Transistor.fm does not assign webhook events. Be awared that, because of this, we can not check the reliability of this event.");
+    this._emit(event.body.data);
+    console.log("Event emitted");
```
[the remark this answers is not in the document]
Comment (vellames) [reply components/transistor_fm/sources/new-event/new-event.mjs:62]: Yeah. Just to inform that Transistor does not assign the call

File: components/transistor_fm/common/options.mjs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export default {
```
[the remark this answers is not in the document]
Comment (vellames) [reply components/transistor_fm/common/options.mjs]: Sure

**Review by alysonturing — APPROVED** [review]
LGTM!

**Pull request merged**